presets/presetManager.py

```diff
@@ -0,0 +1,272 @@
+import os.path
+import logging
+import json
+
+logger = logging.getLogger(__name__)
+
+
+class PresetManager(object):
+    '''PresetManager deals with presets loading, validating, storing
+
+    you can use it like this:
+
+    python::
+    pm = PresetManager(["/path/to/presets/folder", "/another/path"])
+    '''
+
+    MAX_DEPTH = 2
+
+    def __init__(self, paths, strict=False):
+        """load, validate, and store presets under ```self.presets```
+
+        :param paths:  list of paths to preset file
+                       or folder containing presets.
+        :param strict: if strict is False
+                       all exceptions and errors
+                       will be converted to log messages.
+        """
+
+        self.presets = {}
+        self.strict = strict
+
+        # normalize paramter type string -> array
+        if isinstance(paths, basestring):
+            paths = [paths]
+
+        self._loadPaths(paths)
+
+    def _loadPaths(self, paths, depth=0):
+        '''
+        Goes recursevly through the given list of paths
+        in order to find and pass all preset files to ```_loadPreset()```
+        '''
+
+        if depth > self.MAX_DEPTH:
+            return
+
+        for p in paths:
+            path = p
```

Comment:
For strings seems to be different:

```python
import os

l = ["","/tmp/../tmp/","/home///ael/./././"]
print l
for p in l:
    path = p
    path = os.path.normpath(path)
print l

for p in l:
    p = os.path.normpath(p)
print l
```
Both for cycles do not modify objects in ```l```.